index.js

```diff
@@ -171,6 +181,19 @@ class KeyringController extends EventEmitter {
     this.keyrings = await this.unlockKeyrings(password);
     this.setUnlocked();
     this.fullUpdate();
+
+    return this.encryptedKey;
+  }
+
+  /* 
+    MV3:  Unlock by encrypted key
+  */
+  async submitEncryptedKey(encryptedKey) {
+    this.keyrings = await this.unlockKeyrings(undefined, encryptedKey);
+    this.setUnlocked();
+    this.fullUpdate();
```

Comment:
I don't know what this part is doing, assuming some signaling that's not relevant to the security or lock/unlock process.